docs/developing.md

```diff
@@ -158,37 +205,120 @@ Navigate to the `python3` executable, for conda installed via homebrew this may
+### Adding folders to Python path
 
 For some projects you may need to add directories to your PYTHONPATH to make packages importable.
 For example, `caikit/caikit` has a `sample_lib` test fixture which contains a sample implementation of an AI library using caikit.
 The top-level `conftest.py` will take care of adding that fixture to the syspath, but if you want PyCharm to have code-completion and indexing support, you'll need to configure it.
 
+#### PyCharm
+
 To do this in PyCharm, right-click the directory in the project explorer and select `Mark Directory As > Sources Root`
 
 ![](../assets/images/mark_sources.png)
 
-### Running a debugger with PyCharm
+#### VS-Code
+
+Create a `.env` file in the root directory, and add things to that:
+
+```
+PYTHONPATH=tests/fixtures
+```
+
+### Running a debugger
+
+#### With PyCharm
 
 If PyCharm's environment has been configured correctly, then debugging is pretty straightforward.
 Breakpoints are set by clicking on the left editor sidebar next to the line numbers. Conditions can be added to breakpoints by right-clicking on them.
 
 Debugging can be run by either:
+
 1. Starting a debug session locally: Grab a unit test or `__main__` entrypoint and click the green run flag next to it
 2. Attaching a debugger to a running process: Start a process (e.g. `$ python3 -m caikit.runtime`) and then select `Run > Attach to Process`
 
 Remote debugging is available in PyCharm Professional. If you pay for that, you probably know how to use it!
+
+#### With VS-Code
+
+ow to debug python modules is covered in this article: https://code.visualstudio.com/docs/python/debugging
+
+##### TL;DR
+
+1. Create a `.vscode` directory right next to `Caikit`'s home directory, and put this `launch.json` file within it:
+
+   ```
+   {
+   "version": "0.0.1",
+   "configurations": [
+       {
+       "name": "Python: Remote Attach",
+       "type": "python",
+       "request": "attach",
+       "connect": { "host": "127.0.0.1", "port": 3000 },
+       "justMyCode": false
+       }
+   ]
+   }
+   ```
+
+   Note: `justMyCode` is a very cool feature that enables you to debug through external dependencies as well!
+
+2. Once you do that, you need to install `debugpy` (Source code link: https://github.com/microsoft/debugpy):
+
+   ```
+   pip install debugpy
+   ```
+
+3. You need to start `Caikit` through debugpy in order to activate the debugger:
+
+   ```
+   python3 -m debugpy --listen localhost:3000 --wait-for-client -m caikit.runtime
+   ```
+
+   This should wait for the user to start the debug session through `vs-code` and it should stop on any breakpoints that you have configured.
+
+## Sample lib
+
+### What is it
+
+The `sample_lib` is a very simple non-production Caikit library that exists within the `tests/fixtures/sample_lib` directory, and consists of `Modules` and `Tasks` like a production `Caikit` library but with boilerplate code within those `Modules` . This library was created only for helping with unit testing.
```

Comment:
Should we link out here? Or provide more context that this is in reference to the `caikit/caikit` repo?

Reply:
Just a link to the directory you mean?

Reply:
Added a link